### DISCUSSION TITLE
the modeling of converter

### DISCUSSION THREAD

**Discussion by nummb** [Aalto-Electric-Drives/motulator#97] [Q&A] [answered]
Hello sir,
    About the modeling of converter, is it possible to do more accurate modeling, such as the voltage drop of diode and the dead time.

**Comment by mhinkkan** [comment, marked as answer]
Thanks, @nummb, for your relevant question and interest in motulator. 

It would be possible to model the dead-time effect and the power semiconductor voltage drops, but we haven't yet implemented these effects in motulator. Even if parasitic capacitances of a converter are omitted, these effects are quite tricky to model due to the zero-current clamping phenomenon, see Section II here:

J.-W. Choi and S. -K. Sul, "A new compensation strategy reducing voltage/current distortion in PWM VSI systems operating with low output voltages," IEEE Trans. Ind. Appl., vol. 31, no. 5, pp. 1001-1008, Sep./Oct. 1995, https://doi.org/10.1109/28.464512

Implementing such an advanced converter model would require careful planning to ensure its modularity and clarity. This is a challenging topic, but contributions are welcome! 

By the way, we are currently working on a multi-level converter model. I believe we could add it to the public repo in the coming weeks or months.

**Reply by nummb** [reply]
Thank you @mhinkkan for your patient reply. And motulator is a great project.

Recently I was doing a simulation of the initial rotor position estimation of PMSM by injecting modulated pulse voltage vectors. And I am confused about "Universal Bridge" in simulink. So I wonder if there is a clearer formulation.

And then, many thanks for your open source projects.
